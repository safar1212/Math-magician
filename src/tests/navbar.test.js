import Navbar from "../components/navBar";
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';


describe('Navbar', () => {

  test('Builds the snapchot of the navbar component safely', () => {
    const navbar = renderer.create(<Router><Navbar /></Router>).toJSON();
    expect(navbar).toMatchSnapshot();
  });

  test('renders navbar component', () => {
    render(<Router><Navbar /></Router>);

    screen.getByRole('navigation');
  });

});




