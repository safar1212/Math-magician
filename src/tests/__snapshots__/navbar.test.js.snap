// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Builds the snapchot of the navbar component safely 1`] = `
<nav
  className="navBar"
>
  <h1>
    Math Magician
  </h1>
  <ul>
    <li>
      <a
        className=""
        href="/home"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
